packages/sdk-hooks/src/useSyncChain/useSyncChain.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { renderHook } from "@testing-library/react";
+import { describe, test, expect, vi, beforeEach } from "vitest";
+import { useSyncChain } from "./useSyncChain";
+
+import { anvil, optimismSepolia } from "viem/chains";
+
+const mocks = vi.hoisted(() => {
+  return {
+    useAccount: vi.fn(),
+    useSwitchChain: vi.fn(),
+    useChainId: vi.fn(),
+    useChains: vi.fn(),
+  };
+});
+describe("useChain with anvil", () => {
+  beforeEach(() => {
+    vi.mock("wagmi", () => ({
+      ...mocks,
+    }));
+
+    mocks.useAccount.mockReturnValue({
+      address: "0x123",
+      chainId: anvil.id,
+    });
+    mocks.useSwitchChain.mockImplementation(() => {
+      return {
+        switchChain: vi.fn(),
+      };
+    });
+    mocks.useChainId.mockReturnValue(anvil.id);
+    mocks.useChains.mockReturnValue([optimismSepolia, anvil]);
+  });
+
+  test("env and wallet chain match", () => {
+    const chainFromEnv = "anvil";
+
+    const { result } = renderHook(() => useSyncChain(chainFromEnv));
+    const chain = result.current.chain;
+
+    expect(result.current.error).toBeNull();
+    expect(chain).toBeDefined();
+    expect(chain?.name).toBe("Anvil");
```

Comment:
A more future-proof assertion would be to check returned chain id which has minimal risk of ever changing while naming may change in wagmi to be lowercase or whatnot.

Reply:
indeed kind sir